### ISSUE TITLE
Extend camViz (C++) for ROS2 

### ISSUE THREAD

**Issue by pankhurivanjani** [JdeRobot/viz#35]
Using the drivers and tool developed in 

1. [image-tool](https://github.com/TheRoboticsClub/colab-gsoc2019-Pankhuri_Vanjani/tree/master/camera-interface)

2. [python-scripts for running image tool](https://github.com/TheRoboticsClub/colab-gsoc2019-Pankhuri_Vanjani/tree/master/image-tool-scripts)

refactor camViz tool

**Comment by pankhurivanjani** [comment]
I encountered an issue during refactoring this tool. 
In camviz, viewer.cpp file uses  this header

#include <resourcelocator/gladelocator.hpp> 

When I Check my system this file can be found in /opt/jderobot/include/resourcelocator. While compiling I get issues like
```
fatal error: resourcelocator/gladelocator.hpp: No such file or directory
 #include <resourcelocator/gladelocator.hpp>
```
My doubt here is can we directly use the header files similar to gladelocator.hpp which are present in installed jderobot system in a new project.. maybe... after using source /opt/jderobot/setup.bash for building that project? Or... if something else also needs to be done other than linking files in CMakeList?

P.S. I got a temporary solution for this for some header files by using the exact path of the header file like  #include </opt/jderobot/include/resourcelocator/gladelocator.hpp>

P.P.S My current CMakeLists.txt can be found in [camviz-CMakeLists.txt](https://github.com/pankhurivanjani/viz/blob/pvvizros1/camViz/CMakeLists.txt)

**Comment by fqez** [comment]
Hi @pankhurivanjani 

hmm I am not sure how we should approach this issue... I mean, the best solution would be to refactor the GUI so it doesn't depends on a glade file, but I think that's too much work. 

Another approach to try to isolate camViz from JdeRobot/base would be to copy its glade file inside its directory, so you can specify the location of that glade file and stop worrying about external programs (such as gladelocator). This one is particulary tricky because, as I see in your CMakeLists, you depend on other stuff from JdeRobot base such as jderobottypes, jderobotconfig and resourcelocator.

Have you tried using the GLADE_DIR variable [here](https://github.com/pankhurivanjani/viz/blob/30733965806e2dca1b629dc0c40572ba97d8c4b6/camViz/CMakeLists.txt#L10) ? It may work, but I don't remember exactly how.

If you think that's a lot of work, you can simply leave the absolute path in you include sentence for now, and I will take care of the refactor any time soon.

What do you think?

**Comment by pankhurivanjani** [comment]
Hi @fqez

1. Leaving absolute path worked for stuff like jderobottypes but not for resourcelocator
I got the following; they are due to path errors in in directory  resourcelocator whichhas header files dependent on each other. 
```
In file included from /home/pankhuri/week8-camviz/viewer.cpp:26:0:
/home/pankhuri/week8-camviz/resourcelocator/gladelocator.hpp:24:10: fatal error: resourcelocator/resourcelocator.hpp: No such file or directory
 #include <resourcelocator/resourcelocator.hpp>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
CMakeFiles/camViz.dir/build.make:86: recipe for target 'CMakeFiles/camViz.dir/viewer.cpp.o' failed
make[2]: *** [CMakeFiles/camViz.dir/viewer.cpp.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/camViz.dir/all' failed
make[1]: *** [CMakeFiles/camViz.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```
2.  I enabled the  GLADE_DIR variable in [this](https://github.com/pankhurivanjani/viz/blob/30733965806e2dca1b629dc0c40572ba97d8c4b6/camViz/CMakeLists.txt#L10) but I don't know how is it going to help.

3. Yes, I copied the glade file in isolated camViz. so maybe I can get rid of gladelocator for now atleast.

**Comment by pankhurivanjani** [comment]
Solved by [#39 ](https://github.com/JdeRobot/viz/pull/39)
